formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -115,38 +125,33 @@ internal class FormulaManagerImpl<Input, State, Output>(
             }
 
             result = evaluation
+            if (!isValidationEnabled) {
+                inspector?.onEvaluateFinished(type, evaluation.output, evaluated = !allEvaluationsSkipped)
+            }
         }
         isEvaluating = false
-
         return result
     }
 
+    private data class EvaluationResult<Output>(val evaluation: Evaluation<Output>, val skipped: Boolean)
+
     /**
      * Creates the current [Output] and prepares the next frame that will need to be processed.
      */
-    private fun evaluation(input: Input, transitionID: Long): Evaluation<Output> {
+    private fun evaluation(input: Input, transitionID: Long): EvaluationResult<Output> {
```

Comment:
I'm pretty sure there's a cleaner way to represent this logic, but I'm not confident enough to make too many assumptions. I'm thinking maybe the transitionID could be checked  instead of porting EvaluationResult.skipped.